core/src/test/java/org/elasticsearch/cluster/metadata/IndexNameExpressionResolverTests.java

```diff
@@ -643,6 +644,51 @@ public void testConcreteIndicesWildcardWithNegation() {
         assertEquals(0, indexNames.length);
     }
 
+    // concreteIndexNames must be able to resolve the provided wildcard only against the defined
+    // indices when ignoreAliases option is set or against the defined indices and aliases otherwise
```

Comment:
I tend to think that this comment is not needed. Maybe add a small comment close to where you create the two indices options?